runtime/compute/cker/include/cker/operation/DynamicUpdateSlice.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NNFW_CKER_DYNAMIC_UPDATE_SLICE_H__
+#define __NNFW_CKER_DYNAMIC_UPDATE_SLICE_H__
+
+#include "cker/Shape.h"
+
+#include <vector>
+
+namespace nnfw::cker
+{
+
+template <typename T>
+void UpdateSlice(int32_t current_dim, int32_t max_dim, const std::vector<int32_t> output_stride,
+                 const std::vector<int32_t> update_stride, const Shape &update_shape,
+                 const T *update, const std::vector<int64_t> indices_data, T *output)
+{
+  if (current_dim == max_dim)
+    return;
+
+  if (current_dim == max_dim - 1)
+  {
+    output += indices_data[current_dim] * output_stride[current_dim];
+    memcpy(output, update, update_shape.Dims(max_dim - 1) * sizeof(T));
+  }
+  else
+  {
+    output += indices_data[current_dim] * output_stride[current_dim];
+    for (int i = 0; i < update_shape.Dims(current_dim); ++i)
+    {
+      UpdateSlice(current_dim + 1, max_dim, output_stride, update_stride, update_shape, update,
+                  indices_data, output);
+      output += output_stride[current_dim];
+      update += update_stride[current_dim];
+    }
+  }
```

Comment:
I noticed that there is a potential risk. If the values in `indices_data` go beyond the boundaries of the input shape, output and update might point to invalid memory regions, which could lead to out-of-bounds access.
To make the code safer, I recommend making sure that all indices are properly clamped before they are used, or also adding checks to confirm that both the indices and stride calculations are correct at runtime.

Reply:
I'll update to use class and private method to hide `UpdateSlice`.